pallets/pools/src/lib.rs

```diff
@@ -1524,20 +1607,57 @@ pub mod pallet {
 
 					pool.tranches.combine_with_mut_residual_top(
 						tranches.into_iter(),
-						|tranche, (new_tranche_type, seniority)| {
+						|tranche, tranche_update| {
 							// Update debt of the tranche such that the interest is accrued until now with the previous interest rate
 							tranche.accrue(now)?;
 
-							tranche.tranche_type = new_tranche_type.clone();
+							tranche.tranche_type = tranche_update.tranche_type.clone();
 
-							if let Some(new_seniority) = seniority {
+							if let Some(new_seniority) = tranche_update.seniority {
 								tranche.seniority = new_seniority.clone();
 							}
+
 							Ok(())
 						},
 					)?;
 				}
 
+				// Checking if we have a new tranche?
```

Comment:
I checked via the UI, and it seems that we just send over the updated or new tranches. Would need a hint on how a new tranche is marked or if we indeed get all current tranches in the pool (updated + new) from the call. Maybe @offerijns @branan @mustermeiszer know more?

Reply:
We currently do not support adding or removing of tranches. So the vector can be mapped one-to-one to the existing tranches.

If we wanna support adding, removing, inserting tranches, we need to change more.